### PR TITLE
Update

### PR DESCRIPTION
**Pull request by Nearata**

### CODE REVIEW

**Comment by jslirola** [comment]
Hi Nearata, as discussed here: https://discuss.flarum.org/d/27959-prevent-double-posting/23

This is not working for me. How can I provide you with more information about my problem? These days I will try to use xDebug or debugger in the browser to follow the process.

**Comment by Nearata** [comment]
@jslirola is the whole update not working or just the alert not showing up?

**Comment by Nearata** [comment]
@jslirola please lemme know if this still happens, iam still trying to figure out what it could be

**Comment by jslirola** [comment]
@Nearata Great! You already fixed the issue with the button but the alert is still missing.

double_posting_17

I will keep testing the extension to give you more details as soon as possible.

**Comment by jslirola** [comment]
Okay, some news after testing with different values on Time Limit.

When I set 0 I can't see the alert and I'm getting error 403 because I was the last poster.

When I set 1 I can see the alert and I'm forced to edit my last reply so that's good, but once the minute passes, I don't see the alert anymore and I can post again without problem or errors even being the last one to have written.

**Comment by Nearata** [comment]
looking at the previous code, it should be

0 -> wait until someone else replies
others -> wait some time before allowed to reply again

also, there should be another alert, outside the composer body, that shows up only when you arent allowed to edit your previous post and also not allowed to double post.

**Comment by jslirola** [comment]
Finally it's working properly using 0 as time limit.

Two small details more in case you want to improve your code:

- Once you click on Reply button, the composer body is opened with the last reply but it includes a mention of the message itself. I don't know if you want to delete it or not.

imagen

- Where is that alert you're talking about? Do I need to change the permissions to restrict post editing or something? I haven't seen it yet, but maybe I'm missing something.

Thank you for your effort and time to fix this.

**Comment by jslirola** [comment]
@Nearata After these weeks, I thought it was working fine but a user reported to me that he couldn't reply in a discussion with the message that appears in this extension. The user who had written for the last time was me and not him. Any idea what happened? I have disabled it until I find the problem.

**Comment by jslirola** [comment]
imagen

It seems that it keeps crashing, maybe I can try using the debugger and find the problem, but I need some hints. The error of the extension is this, although in the capture it appears translated by me:

```
You are the last poster of this discussion.
```

**Comment by Nearata** [comment]
Oh there it is, the alert i was talking about before when you cant reply and not allowed to edit your last post.

Does it shows up even after a page reload?

**Comment by jslirola** [comment]
Yes, also I cleared cache several times and tried to use the web browser debugger, but couldn't make any progress, as all the JS code was minified.



Do you have a Discord account? Maybe we can go through it together from there and I can share my screen to show you all the details.

**Comment by Nearata** [comment]
> Yes

I have

Time Limit = 0
Allow post editing = 10 minutes

and works as expected.

steps:

user1 reply

- waits


user2 replies

- user1 can reply again, no longer the last poster

user1 replies

> Do you have a Discord account?

Pytey#4345

**Comment by jslirola** [comment]
I'll update the extension tomorrow with your latest change and we'll see if it works.

Btw, I sent you a friend request on Discord since I can't message you if we don't have a common group or something.

Thank you.

**Comment by jslirola** [comment]
It didn't work.

**Comment by Nearata** [comment]
no idea, i cant reproduce this issue, not even before the policy fix

**Comment by Nearata** [comment]
ok, it _should_ work now, is like before me coming here lol

**Comment by jslirola** [comment]
Great! I will test it today or tomorrow. If it doesn't work, I'll look for a way to provide more logs or screenshots that help find the root cause of the problem :)

**Comment by jslirola** [comment]
Unfortunately it doesn't work yet. But I have discovered something new that can help to discover the reason.

If I try to reply with a user who wasn't the last to write, I get the error popup and it doesn't open composer, however, if I try to write to a discussion where I'm the last one, composer opens and tells me that I I was last and I need to edit my answer instead of writing one. Screenshot:

imagen

In short, you can only write in which a user himself is the last to reply instead of any other discussion.

**Comment by jslirola** [comment]
It seems to keep working wrong. Talk to me on Discord and we'll try to see the details there.

**Comment by jslirola** [comment]
It finally works! I'm going to activate it on my production server and will let you know in the next few weeks if I find any other errors. In case I don't comment again in that period of time, I think it will be ready to merge :)
